python/lsst/display/firefly/footprints.py

```diff
@@ -181,20 +189,20 @@ def createFootprintsTable(catalog, xy0=None,
                                'footprint_corner2_x;footprint_corner2_y;spans;peaks'))
     outTable.infos.append(Info(name='pixelsys', value='zero-based'))
     # Check whether the coordinates are included and are valid
-    if (('coord_ra' in sourceTable.colnames) and
-            ('coord_dec' in sourceTable.colnames) and
-            np.isfinite(sourceTable['coord_ra']).any() and
-            np.isfinite(sourceTable['coord_dec']).any()):
-        coord_column_string = 'coord_rafla;coord_dec;EQ_J2000'
-    elif (('base_SdssCentroid_x' in sourceTable.colnames) and
-            ('base_SdssCentroid_y' in sourceTable.colnames) and
-            np.isfinite(sourceTable['base_SdssCentroid_x']).any() and
-            np.isfinite(sourceTable['base_SdssCentroid_y']).any()):
+    if (('coord_ra' in inputColumnNames) and
+            ('coord_dec' in inputColumnNames) and
+            np.isfinite(outTable.array['coord_ra']).any() and
+            np.isfinite(outTable.array['coord_dec']).any()):
+        coord_column_string = 'coord_ra;coord_dec;EQ_J2000'
+    elif (('base_SdssCentroid_x' in inputColumnNames) and
+            ('base_SdssCentroid_y' in inputColumnNames) and
+            np.isfinite(outTable.array['base_SdssCentroid_x']).any() and
+            np.isfinite(outTable.array['base_SdssCentroid_y']).any()):
         coord_column_string = 'base_SdssCentroid_x;base_SdssCentroid_y;ZERO_BASED'
-    elif (('base_NaiveCentroid_x' in sourceTable.colnames) and
-            ('base_NaiveCentroid_y' in sourceTable.colnames) and
-            np.isfinite(sourceTable['base_NaiveCentroid_x']).any() and
-            np.isfinite(sourceTable['base_NaiveCentroid_y']).any()):
+    elif (('base_NaiveCentroid_x' in inputColumnNames) and
+            ('base_NaiveCentroid_y' in inputColumnNames) and
+            np.isfinite(outTable.array['base_NaiveCentroid_x']).any() and
+            np.isfinite(outTable.array['base_NaiveCentroid_y']).any()):
         coord_column_string = 'base_NaiveCentroid_x;base_NaiveCentroid_y;ZERO-BASED'
     else:
         raise RuntimeError('No valid coordinate columns in catalog')
```

Comment:
It might be more efficient if we did all error checking and modifications on the original table and converted to the VO format at the end.

Reply:
Yes, this check could be made earlier until waiting to test for valid coordinates at the end. We are likely going to make more changes to the columns that are used for the center coordinates of the footprints so this should be kept in mind.